### ISSUE TITLE
WIP UIA text provider breaks Narrator

### ISSUE THREAD

**Issue by yaakovschectman** [flutter/flutter#127125]
UIA, which is currently disabled, has suffered a regression since we last tried to enable it wrt to Windows Narrator. Narrator fails to query the text contained in elements that provide the `UIA_TextPatternId` pattern, and reads "no item in view" instead. This was previously not an issue. I have yet to identify any changes in relevant source files that could have caused this.

NVDA does not suffer from this issue.  
When the text and value patterns are provided, Narrator reads "no item in view".  
When only the value pattern is provided, Narrator reads the value properly, but text-specific features will not be available.
When only the text pattern is provided, Narrator only reads the control type and optional semantics name, and does not read the contained text.

**Comment by yaakovschectman** [comment]
I have found that, when using Narrator, the values of `start()->clone()` and `end()->clone()` obtained on https://github.com/flutter/engine/blob/main/third_party/accessibility/ax/platform/ax_platform_node_textrangeprovider_win.cc#L1127 degenerate to the same value, yielding an empty string. The thing is, when using NVDA, they differ in value and return a non-empty string. Furthermore, the only places I see where the values of the start and end endpoints of the text range can be set or changed are:
 - https://github.com/flutter/engine/blob/main/third_party/accessibility/ax/platform/ax_platform_node_textrangeprovider_win.cc#L1553
 - https://github.com/flutter/engine/blob/main/third_party/accessibility/ax/platform/ax_platform_node_textrangeprovider_win.cc#L1572
 - https://github.com/flutter/engine/blob/main/third_party/accessibility/ax/platform/ax_platform_node_textrangeprovider_win.cc#L1543
 - https://github.com/flutter/engine/blob/main/third_party/accessibility/ax/platform/ax_platform_node_textrangeprovider_win.cc#L1548
And when setting breakpoints on each of them, the breakpoints are tripped and those values are set to the non-degenerate values before `GetString` is called, even when Narrator is in use. I interpret this to mean that either some unsightly memory manipulation is taking place obfuscated by Narrator, or our code sneakily writes to these values somewhere that has hid under the radar.
